decoupled-site/frontend/src/pages/SearchPage.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import Footer from "../components/Footer";
+import Header from "../components/Header";
+import SearchButton from "../components/SearchButton";
+import { SearchQuery, useSearchQuery } from "../generated";
+import { generateGQLSearchQueryVars } from "../helpers/queryCacheHelper";
+import { getImageUrl, isEditOrPreviewMode } from "../helpers/urlHelper";
+
+const singleKeyUrl = process.env.REACT_APP_CONTENT_GRAPH_GATEWAY_URL as string
+
+function SearchPage() {
+    const [token, setToken] = useState("")
+    const [searchParams] = useSearchParams()
+    const modeEdit = isEditOrPreviewMode()
+    let data: SearchQuery | undefined = undefined
+    let sortOption: string = "ASC"
+    let queryString: string | null
+    let resultNumber : number | undefined
+    let variables: any
+    let options: {value: string; key: string}[] = [
+        {value: "ASC", key: "ASC"},
+        {value: "DESC", key: "DESC"}
+    ]
+
+    queryString = searchParams.get("q")
+    variables = generateGQLSearchQueryVars(token, window.location.pathname, queryString, sortOption);
+
+    const { data : searchQueryData } = useSearchQuery({ endpoint: singleKeyUrl }, variables, { staleTime: 2000, enabled: !modeEdit || !!token });
+    data = searchQueryData
+    resultNumber = data?.ArtistDetailsPage?.items?.length ?? 0
+
+    return (
+        <div>
+            <Header />
+            <div className="search-container">
+                <div className="back-button">
+                    <a href={window.location.origin} className="home-link">
+                        <span>Back to Landing page</span>
+                    </a>
+                </div>
+                <div className="search-panel">
+                    <SearchButton />
+                </div>
+                <div className="search-description">
+                    <h6>Your search for <span className="search-term">{queryString}</span> resulted in <span className="search-term">{resultNumber}</span> hits</h6>
+                </div>
+                {/* <div className="search-sorting">
```

Comment:
The sorting function is more complicated than I thought it would be, so I would propose a new story to apply the sorting function, and also some other additional features for this search page.
I created a draft story here: https://jira.sso.episerver.net/browse/CG-3485